src/lib/encoder/Encoder.h

```diff
@@ -24,6 +24,10 @@ class Encoder
 	std::optional<cv::Mat> encode_next(STREAM& stream);
 
 	unsigned encode(const std::string& filename, std::string output_prefix);
+
+	template <typename STREAM>
+	fountain_encoder_stream::ptr create_fountain_encoder(STREAM& stream, int compression_level=6);
```

Comment:
There is still some work to be done with this interface -- but instead of calling `encode_fountain`, one can now do:
```
auto fe = enc.create_fountain_encoder(instream);
auto mat = enc.encode_next(*fe);
```
... usually with the 2nd part in some kind of loop. I'll probably extend this to non-fountain encodes somehow or another.